### ISSUE TITLE
Enhancement: add support for stub files

### ISSUE THREAD

**Issue by OutSquareCapital** [jsh9/pydoclint#303]
It would be very nice if Pydoclint could run on stubs files.

Docstring do need to live in them if your package is fully compiled (which is the case of [my library](https://github.com/OutSquareCapital/pyochain/tree/dev)), but that means I unfortunately can't benefit from pydoclint anymore.

I made a pytest plugin to [run code blocks/doctests in stubs](https://github.com/OutSquareCapital/pytest-docflex/tree/master), and the code is pretty straightforward, so I would *assume* that it should be relatively easy for pydoclint to support them. 

It's just the same AST at the end of the day.

In any case, very nice tool :)

**Comment by jsh9** [comment]
Hi @OutSquareCapital , I'm not quite familiar with stub files, so maybe you could provide more detailed specs on what needs to be implemented for this feature. Thanks!

**Comment by OutSquareCapital** [comment]
Well I raised the issue after seeing that it didn't detected anything besides the `__init__.py` (only "true" python file in the folder) when I runned my usual command, but trying again with specific paths, it does work out of the box:
```shell
PS C:\Users\tibo\python_codes\pyochain> uv run pydoclint pyochain/                
Skipping files that match this pattern: \.git|\.tox
pyochain\__init__.py
🎉 No violations 🎉
PS C:\Users\tibo\python_codes\pyochain> uv run pydoclint pyochain/core/_option.pyi
Skipping files that match this pattern: \.git|\.tox
pyochain\core\_option.pyi
pyochain/core/_option.pyi
    60: DOC502: Method `OptionType.__bool__` has a "Raises" section in the docstring, but there are not "raise" statements in the body
    297: DOC105: Method `OptionType.eq`: Argument names match, but type hints in these args do not match: other
    421: DOC502: Method `OptionType.unwrap` has a "Raises" section in the docstring, but there are not "raise" statements in the body
    443: DOC502: Method `OptionType.expect` has a "Raises" section in the docstring, but there are not "raise" statements in the body
    619: DOC105: Method `OptionType.or_else`: Argument names match, but type hints in these args do not match: f
    619: DOC203: Method `OptionType.or_else` return type(s) in docstring not consistent with the return annotation. Return annotation types: ['Option[T | S]']; docstring return section types: ['Option[T]']
    944: DOC203: Method `OptionType.transpose` return type(s) in docstring not consistent with the return annotation. Return annotation types: ['Result[Option[S], E]']; docstring return section types: ['Result[Option[T], E]']
    1018: DOC605: Class `Some`: Attribute names match, but type hints in these attributes do not match: value  (Please read https://jsh9.github.io/pydoclint/checking_class_attributes.html on how to correctly document class attributes.)
PS C:\Users\tibo\python_codes\pyochain> 
```
So a few path options adjustement would do the trick (or just checking for `.pyi` AND `.py`) .

BTW, it could even work on markdowns out of the box, just by checking for "```" fences (just like I do in my pytest plugin linked above):
```shell
PS C:\Users\tibo\python_codes\pyochain> uv run pydoclint foo.md
Skipping files that match this pattern: \.git|\.tox
foo.md
foo.md
    340: DOC203: Method `SetMut.__isub__` return type(s) in docstring not consistent with the return annotation. Return annotation types: ['SetMut[T]']; docstring return section types: ['int']
PS C:\Users\tibo\python_codes\pyochain> 
```

EDIT:

Another adjustment would be to skip checking the function bodies

**Comment by OutSquareCapital** [comment]
Okay so after some more testing I can confirm everything works already, only one adjustment and optionally a QOL feature would be needed.

1) skip checking function bodies for raise statements

2) add an option, like `--stubs`, for conveniently parsing `.pyi` files on top of `.py` ones. Would avoid not-so intuitive recursive regex pattern uses.
